growthbook/growthbook
# #475 - Allow organizations to configure default behaviour metrics in their organization settings

These changes add the ability for an organization to configure defaults for the following properties:

- Minimum Sample Size
- Max Percentage Change
- Min Percentage change

These are stored in `organization.settings.metricDefaults` in MongoDB.

## Usage

A hook `useOrganizationMetricDefaults()` has been created to always provide default values, either from the organization, or the initial hardcoded values. We should be able to count on these values always being present. Empty org settings or org settings metric defaults will just fall back to the hardcoded default values we were initially using.

## Testing

- Configuring the metric behaviour default values on the organization settings page
- Adding a metric from the /metrics page
- Edit a metric
- See metric stats in the reports/experiments results table
- See metric on the metric detail screen /metric/:mid


## Screenshots

Organization settings page:

<img width="1454" alt="image" src="https://user-images.githubusercontent.com/113377031/190485748-fc665839-b84f-4804-a4cf-dc1f9de8af2e.png">


Metric form:

<img width="862" alt="image" src="https://user-images.githubusercontent.com/113377031/190485116-3f5f512f-5f01-4f80-aaf6-0739a4e5e3f2.png">


Closes https://github.com/growthbook/growthbook/issues/475
